openmetadata-ui/src/main/resources/ui/src/components/NavBar/NavBar.tsx

```diff
@@ -552,37 +580,38 @@ const NavBar = ({
         </div>
 
         <Space align="center" size={24}>
-          <Dropdown
-            className="cursor-pointer"
-            menu={{
-              items: domainOptions,
-              onClick: handleDomainChange,
-              className: 'domain-dropdown-menu',
-              defaultSelectedKeys: [activeDomain],
-            }}
-            placement="bottomRight"
-            trigger={['click']}>
-            <Row data-testid="domain-dropdown" gutter={6}>
-              <Col className="flex-center">
-                <DomainIcon
-                  className="d-flex text-base-color"
-                  height={24}
-                  name="domain"
-                  width={24}
-                />
-              </Col>
-              <Col className="flex-center">
-                <Typography.Text>
-                  {activeDomainEntityRef
-                    ? getEntityName(activeDomainEntityRef)
-                    : activeDomain}
-                </Typography.Text>
-              </Col>
-              <Col className="flex-center">
-                <DropDownIcon height={14} width={14} />
-              </Col>
-            </Row>
-          </Dropdown>
+          <div className="d-flex" ref={domainContainerRef}>
+            <DomainSelectableList
+              hasPermission
+              popoverProps={{ open: isDomainDropdownOpen }}
+              selectedDomain={activeDomainEntityRef}
+              onCancel={() => setIsDomainDropdownOpen(false)}
+              onUpdate={handleDomainChange}>
+              <Row
+                data-testid="domain-dropdown"
+                gutter={6}
+                onClick={() => setIsDomainDropdownOpen(!isDomainDropdownOpen)}>
+                <Col className="flex-center">
+                  <DomainIcon
+                    className="d-flex text-base-color"
+                    height={24}
+                    name="domain"
+                    width={24}
+                  />
+                </Col>
+                <Col className="flex-center">
+                  <Typography.Text>
+                    {activeDomainEntityRef
+                      ? getEntityName(activeDomainEntityRef)
+                      : activeDomain}
+                  </Typography.Text>
+                </Col>
+                <Col className="flex-center">
+                  <DropDownIcon height={14} width={14} />
+                </Col>
```

Comment:
I guess we can use a button instead?

Reply:
DomainSelectableList is wrapping this row inside button only.